entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jSearchRepository.kt

```diff
@@ -19,22 +26,47 @@ class Neo4jSearchRepository(
     override fun getEntities(
         queryParams: QueryParams,
         sub: Option<Sub>,
+        geoQuery: GeoQuery,
         contexts: List<String>
     ): Pair<Int, List<URI>> {
-        val query = if (neo4jAuthorizationService.userIsAdmin(sub))
-            QueryUtils.prepareQueryForEntitiesWithoutAuthentication(queryParams, contexts)
-        else
-            QueryUtils.prepareQueryForEntitiesWithAuthentication(queryParams, contexts)
-
         val userAndGroupIds = neo4jAuthorizationService.getSubjectGroups(sub)
             .plus(neo4jAuthorizationService.getSubjectUri(sub))
             .map { it.toString() }
 
+        val query = if (neo4jAuthorizationService.userIsAdmin(sub))
+            QueryUtils.prepareQueryForEntitiesWithoutAuthentication(queryParams, geoQuery, contexts)
+        else
+            QueryUtils.prepareQueryForEntitiesWithAuthentication(queryParams, geoQuery, contexts)
+
         val result = neo4jClient
             .query(query)
             .bind(userAndGroupIds).to("userAndGroupIds")
             .fetch()
             .all()
-        return prepareResults(queryParams.limit, result)
+
+        if (geoQuery.geoproperty.equals(JsonLdUtils.NGSILD_LOCATION_PROPERTY) &&
+            geoQuery.georel != null &&
+            geoQuery.geometry == "Point" &&
+            geoQuery.coordinates != null
+        ) {
+            val geoResult: ArrayList<Map<String, Any>> = ArrayList()
+            val geo1 = WKTReader().read(geoJsonToWkt(geoQuery.geometry, geoQuery.coordinates.toString()))
+            val georelParams = extractGeorelParams(geoQuery.georel)
+
+            result.forEach {
+                val geo2 = WKTReader().read(it["entityLocation"] as String)
+                val distance = distance(geo1, geo2) * MULTIPLY_DISTANCE
+                if (georelParams.second.equals("<=")) {
+                    if (distance <= georelParams.third!!.toDouble()) geoResult.add(it)
+                } else if (georelParams.second.equals(">=")) {
+                    if (distance >= georelParams.third!!.toDouble()) geoResult.add(it)
+                } else if (georelParams.second.equals("==")) {
+                    if (distance(geo1, geo2) != georelParams.third!!.toDouble()) geoResult.add(it)
+                }
+            }
+            return return prepareResults(queryParams.limit, geoResult)
```

Comment:
I was suggesting to use https://neo4j.com/docs/cypher-manual/current/functions/spatial/#functions-distance, why did you choose another way?

Btw, by filtering the results here, you're breaking the pagination. 

But let's keep it like this for now...

Reply:
I found it easier to treat it like this. 
I couldn't see how to retrieve the longitude and latitude and make the query to calculate the distance at the same time.
The function requires a longitude and a latitude for its operation and I did not see how to do the transformation (wkt to geo) in the same requests